docs/content/configuration.md

```diff
@@ -886,14 +890,19 @@ See [the docs on disk storage](../storage/) for details about the settings.
 ## Server
 
 The `server` configuration sets:
-- the gzip compression settings for `/v0/data`, `/v1/data` and `/v1/compile` HTTP `POST` endpoints
+- for all incoming requests:
+  - maximum allowed request size
+  - maximum decompressed gzip payload size
+- the gzip compression settings for responses from the `/v0/data`, `/v1/data` and `/v1/compile` HTTP `POST` endpoints
 The gzip compression settings are used when the client sends `Accept-Encoding: gzip`
 - buckets for `http_request_duration_seconds` histogram
 
-| Field                                                       | Type        | Required                                                                  | Description                                                                                                                                                                                                               |
-|-------------------------------------------------------------|-------------|---------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `server.encoding.gzip.min_length`                           | `int`       | No, (default: 1024)                                                       | Specifies the minimum length of the response to compress                                                                                                                                                                  |
-| `server.encoding.gzip.compression_level`                    | `int`       | No, (default: 9)                                                          | Specifies the compression level. Accepted values: a value of either 0 (no compression), 1 (best speed, lowest compression) or 9 (slowest, best compression). See https://pkg.go.dev/compress/flate#pkg-constants          |
+| Field | Type| Required | Description |
+| --- | --- | --- | --- |
+| `server.decoding.max_length` | `int` | No, (default: 268435456) | Specifies the maximum allowed number of bytes to read from a request body. |
+| `server.decoding.gzip.max_length` | `int` | No, (default: 536870912) | Specifies the maximum allowed number of bytes to read from the gzip decompressor for gzip-encoded requests. |
```

Comment:
So a max gzip-encoded request could be 256MB? If yes, we should update the description to clarify this. Also I'm not sure what a good default is, how did you come up with that number?

Reply:
> So a max gzip-encoded request could be 256MB? 

The maximum size a gzip *payload* could decompress to would be 256 MB. If you have a lot of redundant strings in your JSON, the raw `Content-Length` header might show only 5 MB for the compressed blob, but the gzip blob could then *decompress* to 256+ MB in size.

The `server.decoding.gzip.max_length` config values allows us to reject payloads that would decompress to ridiculous sizes.

> I'm not sure what a good default is, how did you come up with that number?

I'm not sure what a good default is either! :sweat_smile: I chose to just double the raw `Content-Length` size limit, but I'm very open to other size suggestions.

Reply:
OK the defaults seems high enough to not affect most users.